pkg/db/db.go

```diff
@@ -793,26 +913,475 @@ func (h *DBHandler) RunCustomMigrationAllTables(ctx context.Context, getAllAppsF
+			Deleted:    row.Deleted,
+			Metadata:   resultJson,
+		})
+	}
+	err = closeRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	return envLocks, nil
+}
+
+func (h *DBHandler) DBSelectAllEnvironmentLocks(ctx context.Context, tx *sql.Tx, environment string) (*AllEnvLocksGo, error) {
+	if h == nil {
+		return nil, nil
+	}
+	if tx == nil {
+		return nil, fmt.Errorf("DBSelectAllEnvironmentLocks: no transaction provided")
+	}
+	span, _ := tracer.StartSpanFromContext(ctx, "DBSelectAllEnvironmentLocks")
+	defer span.Finish()
+	selectQuery := h.AdaptQuery(
+		"SELECT version, created, environment, json FROM all_env_locks WHERE environment = ? ORDER BY version DESC LIMIT 1;")
+
+	rows, err := tx.QueryContext(ctx, selectQuery, environment)
+	if err != nil {
+		return nil, fmt.Errorf("could not query all env locks table from DB. Error: %w\n", err)
+	}
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("row closing error: %v", err)
+		}
+	}(rows)
+
+	if rows.Next() {
+		var row = AllEnvLocksRow{
+			Version:     0,
+			Created:     time.Time{},
+			Environment: "",
+			Data:        "",
+		}
+
+		err := rows.Scan(&row.Version, &row.Created, &row.Environment, &row.Data)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil, nil
+			}
+			return nil, fmt.Errorf("Error scanning environment locks row from DB. Error: %w\n", err)
+		}
+
+		//exhaustruct:ignore
+		var resultJson = AllEnvLocksJson{}
+		err = json.Unmarshal(([]byte)(row.Data), &resultJson)
+		if err != nil {
+			return nil, fmt.Errorf("Error during json unmarshal. Error: %w. Data: %s\n", err, row.Data)
+		}
+
+		var resultGo = AllEnvLocksGo{
+			Version:         row.Version,
+			Created:         row.Created,
+			Environment:     row.Environment,
+			AllEnvLocksJson: AllEnvLocksJson{EnvLocks: resultJson.EnvLocks},
+		}
+		err = closeRows(rows)
+		if err != nil {
+			return nil, err
+		}
+		return &resultGo, nil
+	}
+	err = closeRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+
+func (h *DBHandler) DBSelectEnvironmentLockSet(ctx context.Context, tx *sql.Tx, environment string, lockIDs []string) ([]EnvironmentLock, error) {
+	if h == nil {
+		return nil, nil
+	}
+	if tx == nil {
+		return nil, fmt.Errorf("DBSelectEnvironmentLockSet: no transaction provided")
+	}
+	span, _ := tracer.StartSpanFromContext(ctx, "DBSelectEnvironmentLockSet")
+	defer span.Finish()
+	//This is needed as lock ids with '-' might confuse sql and you can pass a string array in query
+	for i, l := range lockIDs {
+		lockIDs[i] = "'" + l + "'" //Assuming no
+	}
+	selectQuery := h.AdaptQuery(
+		"SELECT eslVersion, created, lockID, envName, metadata, deleted" +
+			" FROM environment_locks " +
+			" WHERE envName=? AND lockID in (" + strings.Join(lockIDs, ",") + ")" +
+			" ORDER BY eslVersion DESC " +
+			" LIMIT " + strconv.Itoa(len(lockIDs)) + ";")
+
+	rows, err := tx.QueryContext(ctx, selectQuery, environment)
```

Comment:
the query needs to be getting the latests eslVersion per lock. I think this needs to be done by one SELECT per lockId